db/migrate/20210317094834_change_availability_to_max_assignments.rb

```diff
@@ -0,0 +1,23 @@
+class ChangeAvailabilityToMaxAssignments < ActiveRecord::Migration[6.1]
+  def change
+    add_column :editors,:max_assignments, :integer, null: false, default: 2
+    migrate_availability
+    remove_column :editors, :availability, :string
+  end
+
+  def migrate_availability
+    if Editor.first.respond_to?(:max_assignments) && Editor.first.respond_to?(:availability)
+      map_by_current_availability = { "available" => 2,
+                                      "somewhat" => 1,
+                                      "none" => 0,
+                                      "" => 2,
+                                      nil => 2 }
+      assigned_by_editor = Paper.unscoped.in_progress.group(:editor_id).count
+      Editor.emeritus.update_all(max_assignments: 0)
+      Editor.active.each do |editor|
+        editor.max_assignments = assigned_by_editor[editor.id].to_i + map_by_current_availability[editor.availability]
```

Comment:
I'm not sure this is quite right. I'm currently editing 11 paper, and my max availability is marked as '13' but it shouldn't really be 11 + another 2.

Ideally I think we should assume that the goal of JOSS is to have 2-3 papers assigned to an editor at a time unless we explicitly hear from an editor that they are happy to handle more than this.

Reply:
Yeah, it's a basic initial migration based on the current number of assignments and availability status. Here your max_assignments will be marked as 13 because you are currently editing 11 and available, so we assume you can still take 2 more. In the UI that will be something like: "Editing 11, Max 13"